includes/namespace.php

```diff
@@ -100,18 +149,16 @@ function _print_missing_build_admin_notice() {
  * @return void
  */
 function activate( $network_wide = false ) {
-	if ( version_compare( PHP_VERSION, WEBSTORIES_MINIMUM_PHP_VERSION, '<' ) ) {
-		wp_die(
-		/* translators: %s: PHP version number */
-			esc_html( sprintf( __( 'Web Stories requires PHP %s or higher.', 'web-stories' ), WEBSTORIES_MINIMUM_PHP_VERSION ) ),
-			esc_html__( 'Plugin could not be activated', 'web-stories' )
-		);
-	}
+	global $web_stories_compatibility;
 
-	if ( version_compare( get_bloginfo( 'version' ), WEBSTORIES_MINIMUM_WP_VERSION, '<' ) ) {
+	$web_stories_compatibility->check_php_version();
+	$web_stories_compatibility->check_wp_version();
+	// @todo add in other checks here?
```

Comment:
@swissspidy Thoughts 👆🏻

Reply:
The thing here is that `activate()` is called in two different scenarios:

1. Plugin is activated for a site/network.
1. A new site on the network is created.

Why are we doing these checks for 2)? At that point the plugin is already running - no need to check for the PHP version again. Especially no need to check for all required extensions again.

I feel like we need to split `activate()` into two:

1. One function that only runs on plugin activation.
  * Runs PHP & WP version check
  * Calls the second function.
1. Second function
  * Called by `new_site()` and the above function
  * Inits roles and database upgrader

Reply:
Move logic to it's own function here. https://github.com/google/web-stories-wp/pull/4930/commits/5a0cfa6c4133cd70c818903d9bb5b07ca6f08c34

Does really answer the question regarding what checks should be done here? Should be checking php extensions/class/functions here?

Reply:
Neat!

It needs the bare minimum checks so that the plugin can be activated and the admin notice printed. So the PHP & WordPress version check is enough I think.

Reply:
Agreed, keep it lean. Fixed in https://github.com/google/web-stories-wp/pull/4930/commits/e014421e1108081e2fe7e0927ad60fa7892a4d64